ui/components/app/confirm-data/confirm-data.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+
+import {
+  Color,
+  TextVariant,
+  TEXT_TRANSFORM,
+  TypographyVariant,
+} from '../../../helpers/constants/design-system';
+import { getKnownMethodData } from '../../../selectors';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { useTransactionFunctionType } from '../../../hooks/useTransactionFunctionType';
+
+import Box from '../../ui/box/box';
+import Disclosure from '../../ui/disclosure';
+import TransactionDecoding from '../transaction-decoding';
+import { Text } from '../../component-library';
+
+const ConfirmData = ({ txData, dataComponent }) => {
```

Comment:
since ConfirmData and ConfirmHexData currently don't have plans to be used outside of the ConfirmTransactionBase page component, I think it would be a good idea to add the components to the relevant page directory

Reply:
I would prefer to keep them in `ui/components/app` and not `ui/pages`.